plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -194,17 +202,14 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
         }
 
         if (fileScanIterable == null) {
-            // Used to avoid duplicating work if the Dynamic Filter was already pushed down to the Iceberg API
-            boolean dynamicFilterIsComplete = dynamicFilter.isComplete();
             this.pushedDownDynamicFilterPredicate = dynamicFilter.getCurrentPredicate()
                     .transformKeys(IcebergColumnHandle.class::cast)
                     .filter((columnHandle, domain) -> isConvertableToIcebergExpression(domain));
             TupleDomain<IcebergColumnHandle> fullPredicate = tableHandle.getUnenforcedPredicate()
                     .intersect(pushedDownDynamicFilterPredicate);
             // TODO: (https://github.com/trinodb/trino/issues/9743): Consider removing TupleDomain#simplify
             TupleDomain<IcebergColumnHandle> simplifiedPredicate = fullPredicate.simplify(ICEBERG_DOMAIN_COMPACTION_THRESHOLD);
-            boolean usedSimplifiedPredicate = !simplifiedPredicate.equals(fullPredicate);
-            if (usedSimplifiedPredicate) {
+            if (!simplifiedPredicate.equals(fullPredicate)) {
```

Comment:
nit: I find the comment below confusing:
```
// Pushed down predicate was simplified, always evaluate it against individual splits
```
Do we assume that Iceberg is slow if we didn't simplify predicate?
Do we want to evaluate it per split because of partition columns or data columns pruning?
Why do we set `pushedDownDynamicFilterPredicate` to `all`? (I think it means we actually don't want to push simplified version of it)
Why not always evaluate it against splits and never push?